### ISSUE TITLE
libfdisk: Problem with fdisk_reread_changes after adding logical partition

### ISSUE THREAD

**Issue by vojtechtrefny** [karelzak/util-linux#1112]
Hi, I'm trying to replace libparted by libfdisk in our project and I've found a weird issue with `fdisk_reread_changes` when creating a logical partition immediately after creating the extended one. I'm not sure whether am I doing something wrong or if this is a bug in libfdisk or somewhere else in the system.

Here is a simplified reproducer of the issue: https://gist.github.com/vojtechtrefny/ab2f292ee958c6e28b09831394c404b6

When creating the logical partition the `fdisk_reread_changes` always fails with `Failed to add partition 5 to system` and the partition is not visible in the system. This doesn't happen when adding the partition to an pre-existing extended partition.

From the debug log it looks like the `fdisk_reread_changes` works as expected, detects the change and issues the ioctl call:

```
5564: libfdisk:      TAB: [0x1954df0]: table diff [new table=0x1954700]
5564: libfdisk:      TAB: [0x1954df0]:  scanning old table
5564: libfdisk:      TAB: [0x1954df0]:  #0 UNCHANGED
5564: libfdisk:      TAB: [0x1954df0]: table diff [new table=0x1954700]
5564: libfdisk:      TAB: [0x1954df0]:  scanning old table
5564: libfdisk:      TAB: [0x1954df0]:  scanning new table
5564: libfdisk:      TAB: [0x1954df0]:   initialize to TAB=0x1954700
5564: libfdisk:      TAB: [0x1954df0]:  #4 ADDED
5564: libfdisk:      TAB: [0x1954df0]: table diff [new table=0x1954700]
5564: libfdisk:      TAB: [0x1954df0]:  scanning new table
5564: libfdisk:      TAB: [0x1954df0]: table diff done [rc=1]
5564: libfdisk:     PART: [0x1952120]: #4 calling BLKPG_ADD_PARTITION
5564: libfdisk:      ASK: [0x19546a0]: alloc
5564: libfdisk:      ASK: [0x19546a0]: do_ask for 'warn'

** (process:5564): WARNING **: 14:26:42.252: [fdisk] Failed to add partition 5 to system

5564: libfdisk:      ASK: [0x19546a0]: do_ask done [rc=0]
5564: libfdisk:      ASK: [0x19546a0]: reset
5564: libfdisk:      ASK: [0x19546a0]: free
```

It's not part of the reproducer, but we are also locking the disk using `flock` to keep udev away from it as described [here](https://systemd.io/BLOCK_DEVICE_LOCKING/) but it doesn't help.

Any idea what am I doing wrong?

**Comment by karelzak** [comment]
It seems the problem is that libfdisk adds whole extended partition. That's wrong as kernel assumes only 1 or 2 sectors for extended partitions to avoid overlapping with (nested) logical partitions. We need to fix fdisfdisk_reread_changes().

**Comment by karelzak** [comment]
The bugfix will be available in v2.36.1. Thanks for your report!

**Comment by vojtechtrefny** [comment]
Thank you.
